src/lib.rs

```diff
@@ -311,3 +346,82 @@ impl Write for Device {
         Ok(())
     }
 }
+
+impl Device {
+    pub fn set_mpsse_clock(&mut self, freq: u32) -> std::result::Result<(), io::Error> {
+        const MAX: u32 = 30_000_000;
+        const MIN: u32 = 92;
+
+        assert!(
+            freq >= MIN,
+            "frequency of {} exceeds minimum of {}",
+            freq,
+            MIN
+        );
+        assert!(
+            freq <= MAX,
+            "frequency of {} exceeds maximum of {}",
+            freq,
+            MAX
+        );
+
+        let (divisor, clkdiv) = if freq <= 6_000_000 {
+            (6_000_000 / freq - 1, Some(true))
+        } else {
+            (30_000_000 / freq - 1, Some(false))
+        };
+
+        self.write_all(MpsseCmdBuilder::new().set_clock(divisor, clkdiv).as_slice())?;
+
+        Ok(())
+    }
+}
+
+impl MpsseCmdExecutor for Device {
+    type Error = io::Error;
+
+    fn init(&mut self, settings: &MpsseSettings) -> std::result::Result<(), io::Error> {
+        let millis = u8::try_from(settings.latency_timer.as_millis())
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+
+        if settings.reset {
+            self.usb_reset()
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
```

Comment:
Are you sure mapping all `libftdi` errors to `ErrorKind::Other` is the proper way to go?

Reply:
Not at all. My suggestion to use `io::Error` and `io::ErrorKind::Other` is based on the following two observations:
* `ftdi-rs` errors are not very diverse, in most cases functions either panic in runtime or return `Error::RequestFailed`
* as I mentioned above, for the sake of simplicity I suggest to use the same error type as returned by `io::Read` and `io::Write` implementations in `ftdi-rs`

Maybe it makes sense to add yet another wrapper into src/error.rs, that can be enhanced with more granularity later on based on usage experience ? For instance something along the lines:

```
pub(crate) fn libftdi_to_io(err: Error) -> io::Error {
    io::Error::new(io::ErrorKind::Other, err)
}
```

Reply:
Yes, I think this makes sense, please do.

Reply:
Done